Assets/Scripts/Block2TextConsoleManager.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+namespace MoveToCode
+{
+
+    public class Block2TextConsoleManager : Singleton<Block2TextConsoleManager>
+    {
+        StandAloneInstruction curInstruction;
+     
+
+        public void UpdateConsoleOnSnap() //every time a block is snapped, the console is updated with code/text
+        {
+            Block2TextConsoleManager.instance.ClearConsole();
+            curInstruction = StartCodeBlock.instance.GetMyInternalIArgument() as StandAloneInstruction;
+
+            while (curInstruction != null){
+                Block2TextConsoleManager.instance.AddLine(curInstruction?.DescriptiveInstructionToString());
+                curInstruction = curInstruction.GetNextInstruction();
+            }
+
+        }
+
+            /* <------copied from ConsoleManager------>*/
+        TextMeshProUGUI mainConsole, headerConsole;
+        public static string finishedString = "Code Finished";
+
+        void Awake(){
+            ClearConsole();
+        }
+
+        public TextMeshProUGUI GetHeaderConsole() {
+            if (headerConsole == null)
+            {
+                headerConsole = transform.GetChild(0).GetComponent<TextMeshProUGUI>();
+            }
+            return headerConsole;
+        }
+
+        public TextMeshProUGUI GetMainConsole() {
+            if (mainConsole == null)
+            {
+                mainConsole = transform.GetChild(1).GetComponent<TextMeshProUGUI>();
+            }
+            return mainConsole;
+        }
+
+        public void ClearConsole() {
+            GetMainConsole().text = "";
+        }
+
+        public void SetHeaderText(string t) {
+            GetHeaderConsole().text = t;
+        }
+
+        public void AddLine(string lineToAdd){
+            GetMainConsole().text = string.Join("", GetMainConsole().text, lineToAdd, "\n");
```

Comment:
again my bad code but replace `GetMainConsole().text` with `GetMainText()`

Reply:
GetMainText() = ... is not allowed (LHS of assignment cannot be a function), so I see why you wrote this.